### ISSUE TITLE
Detach the child makes an error

### ISSUE THREAD

**Issue by XavierBoubert** [ractivejs/ractive#2020]
I have an bug but I don't think from Ractive, maybe from Chrome.

I have a *main component* with a *sub-component* in it. When I teardown the *main component* I receive this error:

![error](https://cloud.githubusercontent.com/assets/482251/8434268/b64e4b32-1f4a-11e5-8f09-7302a3500654.png)

The error comes from the ```detach.js``` file.

So I use the _Pause on exception_ button to stop when the error is going to be fired. And I receive this wirred state:

![error-2](https://cloud.githubusercontent.com/assets/482251/8434291/d5327578-1f4a-11e5-97ff-668673e5825a.png)

Basically the Ractive code manages the behavior but I don't understand why Chrome doesn't. Any help?

**Comment by dagnelies** [comment]
@XavierBoubert perhaps the error is in your teardown function. I'd suggest posting it, or, even better, making a simple jsfiddle out of it so people can reproduce/understand the error.

**Comment by XavierBoubert** [comment]
Do you have an example of an error in the teardown function? I'm trying to create a jsfiddle but it's not easy. I'm using [Ractive-Require](https://github.com/XavierBoubert/ractive-require) to require components in demand.

**Comment by fskreuz** [comment]
The following will help even if you don't put up a demo:

- The version of Ractive you use.
- Any transitions you have used.
- Any decorators you have used.
- A piece of your code and surrounding code so we can infer what's happening.

Also, in Chrome, when you pause for exceptions, the exception message will appear in the Scope panel. Make sure the paused exception is the right one, as Ractive causes a lot of managed errors that you might be seeing one of the other errors happening.

In addition, Chrome has some quirks when showing values from source-mapped code. Sometimes, these data don't show up properly in the tooltip, or provide wrong values. Try debugging using the unminified but concatenated version of Ractive just to make sure.

**Comment by XavierBoubert** [comment]
I have found the problem!

It was in the [Ractive-Require](https://github.com/CodeCorico/ractive-require) teardown logic. Now I have fixed that in the [0.5.2](https://github.com/CodeCorico/ractive-require/blob/master/CHANGELOG.md) version.
